src/ImageView.jl

```diff
@@ -148,21 +151,41 @@ end
 function imshow!(canvas::GtkObservables.Canvas,
                  img::AbstractMatrix,
                  annotations::Annotations=annotations())
+    pyr = ImagePyramid(img)
     draw(canvas, annotations) do cnvs, anns
-        copy_with_restrict!(cnvs, img)
+        copy_with_restrict!(cnvs, pyr)
         set_coordinates(cnvs, axes(img))
         draw_annotations(cnvs, anns)
     end
     nothing
 end
 
-function copy_with_restrict!(cnvs, img::AbstractMatrix)
-    imgsz = size(img)
+# cache of the image plus downscaled versions
+ImagePyramid(img::AbstractMatrix) = Any[img]
```

Comment:
Since this is a function, not a type, it should be lowercase.

ImageView supports `n`-dimensional images, so perhaps this should be generalized beyond `AbstractMatrix`? Or does this only happen after slice-selection? If a user is browsing a large 3d array, would the entire pyramid be recomputed each time one changes slice planes?